File: it/test-basic/expected/java/test/Bar.java
package test;

import java.util.Objects;
import java.util.Optional;

public class Bar {
  private final Bar.Inner field;

  public Bar(
    final Bar.Inner field
  ) {
    Objects.requireNonNull(field, "field");
    this.field = field;
  }

  public Bar.Inner getField() {
    return this.field;
  }

  @Override
  public int hashCode() {
    int result = 1;
    result = result * 31 + this.field.hashCode();
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Bar)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Bar o = (Bar) other;

    if (!this.field.equals(o.field)) {
      return false;
    }

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Bar");
    b.append("(");
    b.append("field=");
    b.append(this.field.toString());
    b.append(")");

    return b.toString();
  }

  public static class Builder {
    private Optional<Bar.Inner> field = Optional.empty();

    public Builder field(final Bar.Inner field) {
      this.field = Optional.of(field);
      return this;
    }

    public Bar build() {
      final Bar.Inner field = this.field.orElseThrow(() -> new RuntimeException("field: is required"));

      return new Bar(field);
    }
  }

  public static class Inner {
    private final String field;

    public Inner(
      final String field
    ) {
      Objects.requireNonNull(field, "field");
      this.field = field;
    }

    public String getField() {
      return this.field;
    }

    @Override
    public int hashCode() {
      int result = 1;
      result = result * 31 + this.field.hashCode();
      return result;
    }

    @Override
    public boolean equals(final Object other) {
      if (other == null) {
        return false;
      }

      if (!(other instanceof Inner)) {
        return false;
      }

      @SuppressWarnings("unchecked")
      final Inner o = (Inner) other;

      if (!this.field.equals(o.field)) {
        return false;
      }

      return true;
    }

    @Override
    public String toString() {
      final StringBuilder b = new StringBuilder();

      b.append("Inner");
      b.append("(");
      b.append("field=");
      b.append(this.field.toString());
      b.append(")");

      return b.toString();
    }

    public static class Builder {
      private Optional<String> field = Optional.empty();

      public Builder field(final String field) {
        this.field = Optional.of(field);
        return this;
      }

      public Inner build() {
        final String field = this.field.orElseThrow(() -> new RuntimeException("field: is required"));

        return new Inner(field);
      }
    }
  }
}

File: it/expected/tests/basic/csharp/Test/Bar.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
  [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
  public class Bar {
    /// <summary>
    /// The inner field.
    /// </summary>
    [JsonProperty("field", Required = Required.DisallowNull)]
    public Bar.Inner field {
      get;
    }

    [JsonConstructor]
    public Bar(
      [JsonProperty("field", Required = Required.DisallowNull)] Bar.Inner field
    ) {
      this.field = field;
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      result = result * 31 + this.field.GetHashCode();
      return result;
    }

    public override Boolean Equals(Object other) {
      Bar o = other as Bar;

      if (o == null) {
        return false;
      }

      if (!this.field.Equals(o.field)) {
        return false;
      }

      return true;
    }

    public override String ToString() {
      StringBuilder b = new StringBuilder();

      b.Append("Bar");
      b.Append("(");
      b.Append("field=");
      b.Append(this.field);
      b.Append(")");

      return b.ToString();
    }

    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    public class Inner {
      /// <summary>
      /// The field.
      /// </summary>
      [JsonProperty("field", Required = Required.DisallowNull)]
      public String field {
        get;
      }

      [JsonConstructor]
      public Inner(
        [JsonProperty("field", Required = Required.DisallowNull)] String field
      ) {
        this.field = field;
      }

      public override Int32 GetHashCode() {
        Int32 result = 1;
        result = result * 31 + this.field.GetHashCode();
        return result;
      }

      public override Boolean Equals(Object other) {
        Inner o = other as Inner;

        if (o == null) {
          return false;
        }

        if (!this.field.Equals(o.field)) {
          return false;
        }

        return true;
      }

      public override String ToString() {
        StringBuilder b = new StringBuilder();

        b.Append("Inner");
        b.Append("(");
        b.Append("field=");
        b.Append(this.field);
        b.Append(")");

        return b.ToString();
      }
    }
  }
}
